src/gfp_generic.cpp

```diff
@@ -0,0 +1,198 @@
+
+#include <gfp_generic.h>
+#include <constants.h>
+#include <gfp.h>
+
+namespace bn256 {
+
+    typedef std::array<uint64_t, 8> uint64_array_8_t;
+    typedef std::array<uint64_t, 18> uint64_array_18_t;
+    typedef std::array<uint64_t, 32> uint64_array_32_t;
+
+    static void gfp_carry(gfp& a, uint64_t head){
+        gfp b{};
+        uint64_t carry = 0;
+        for (const auto& pi : p2) {
+            const auto i = &pi - &p2[0];
```

Comment:
Why not just `for (auto i = 0; i < p2.size(); ++i)` and replace `p2` to `p2[i] `below?

Reply:
Done. commit: https://github.com/AntelopeIO/bn256/pull/1/commits/58847c2d9049688f99b05380bafc3863c8150894